### ISSUE TITLE
pipenv path and executable path issues

### ISSUE THREAD

**Issue by datatalking** [shacker/django-todo#140]
I had used the Django-todo years ago and it worked great since @shacker you helped me solve [issue#36](https://github.com/shacker/django-todo/issues/36) #37, #38 the challenge is since 2018 all of what I've worked on is heavy NLP and many different packages I can only get them to work with anaconda.

What I've been working on is a way (potentially) to save your list to a database and it learns which items you prefer over time and attempts to automate the order of the todo list, and I've come up with a way to have it be automated and user selectable for up to eight different variables. So if you have something with a due date of today that item would show up on the list before everything due tomorrow, or if you had 'Project BBQ' and it was dependent upon something like 'Buy BBQ' it will only let you start once that constraint is satisfied But... I'm running into environment stumbling blocks.

Step five is as far as I got on this install and I'm hesitant to mess with my 3.8 environments as I just got a bunch of stuff working and very hesitant to step into the PATH and ENVIRONMENT minefield unless the steps are well understood.

Options to install this with conda or?

TLDR: 100% of my code for the last 5 years has only been written using 'conda install <package>' and its now smooth as glass for Pycharm installs, like my Mac Nvidia GPU with 2800 cores which is near impossible to get working is lighting quick. What is the install option for conda users?

I could help write something or create a patch of some kind, help write new instructions as these don't work in my data science environments.

This first line is just git cloning pipenv itself, ok.
1. `pip3 install --user git+https://github.com/pypa/pipenv.git`

I normally install in `Users/username/sandbox/PycharmProjects/<project_name>`
2. `cd ~/Users/username/sandbox/PycharmProjects/`

It tells me I don't have permission? Which is weird since its my own laptop `fatal: could not create work tree dir 'gtd': Operation not permitted`
3. `git clone git@github.com:shacker/gtd.git`

If I go my normal route and create the environment in conda with the 'gtd' name I can get to this point and open the directory
4. `cd gtd`

I tried to complete this step but it tells me pipenv is incompatible with error message `⠋ModuleNotFoundError: No module named 'virtualenv.activation.xonsh'`, stack overflow says to `pip uninstall virtualenv` I stopped here.
5. `pipenv --python 3.9`   # Initializes the virtual environment
5.1 Possible Fix 'pip uninstall virtualenv' generates a warning message, and I can't have it removing my 3.8 packages as I'm still migrating 3.7 over and that's going to take another 6 months.
`
Uninstalling virtualenv-20.19.0:
  Would remove:
>>>    /Users/username/.local/bin/virtualenv
>>>    /Users/username/.local/lib/python3.8/site-packages/virtualenv-20.19.0.dist-info/*
>>>    /Users/username/.local/lib/python3.8/site-packages/virtualenv/*
`

5.2 While installing this I got an error that might help about how I was installing it and it said it wasn't on PATH ` WARNING: The script virtualenv-clone is installed in '/Users/username/.local/bin' which is not on PATH.`

5.3 On a related note I worked on getting Django-Gentella install working in my .env file working last weekend and struggled with that as well but then I thought if you knew my PATH for related issues might be helpful in diagnosing this?

`
Gentella_PATH:

> project_path = '~Users/username/sandbox/PycharmProjects/django-gentelella '
 
> conda env path = '/Users/username/opt/anaconda3/envs/django-gentelella/bin/python'

> add interpreter = '/Users/username/opt/anaconda3/condabin/conda'

`

# ONHOLD
6. 'pipenv install --dev'  # Installs all dependencies

# ONHOLD
7. 'pipenv shell'  # Activates the environment

I've also been digging around the interwebs and found these steps from their github repo trying to help another conda user. https://github.com/pypa/pipenv/issues/699.

**Comment by gizquier2** [comment]
with all due respect, this does not seem like an issue.
https://realpython.com/python-virtual-environments-a-primer/

**Comment by shacker** [comment]
Thanks @gizquier2 , I agree. 

@datatalking I use neither Anaconda nor pipenv, so I can't be much help there. There is a Discussion section open above where you can have generalized discussions or make support requests. And there's good old Stack Overflow.  But it doesn't sound like you're identifying any particular quirk in Django-todo here.

**Comment by girishc24** [comment]
For setting up Django-todo in a Conda environment, consider installing Pipenv via Conda, cloning the repository, creating and activating a Conda virtual environment, and then installing dependencies using Conda or Pipenv. Make sure to activate your Conda environment throughout the process. If challenges arise, adjusting dependencies or seeking assistance from the project's maintainers may be necessary.

Source:

Conda installation of Pipenv: https://anaconda.org/conda-forge/pipenv

**Comment by shacker** [comment]
This issue is years old with no further movement. Everything is fine with normal pip or uv installation environments. Closing issue, sorry.
